src/Functions/idna.cpp

```diff
@@ -140,59 +140,82 @@ using FunctionIdnaDecode = FunctionStringToString<IdnaDecode, NameIdnaDecode>;
-        .arguments={{"str", "Input string"}},
-        .returned_value={"Returns a unicode-encoded domain name.", {"String"}},
-        .examples={
-            {"simple",
-            "SELECT idnaDecode('xn--strae-oqa.xn--mnchen-3ya.de') AS unicode;",
-            R"(
-┌─unicode───────────┐
-│ straße.münchen.de │
-└───────────────────┘
-            )"
-            }},
-        .category = FunctionDocumentation::Category::Encoding
-    });
+    FunctionDocumentation::Description description_encode = R"(
+Returns the ASCII representation (ToASCII algorithm) of a domain name according to the [Internationalized Domain Names in Applications](https://en.wikipedia.org/wiki/Internationalized_domain_name#Internationalizing_Domain_Names_in_Applications) (IDNA) mechanism.
+The input string must be UTF-encoded and translatable to an ASCII string, otherwise an exception is thrown.
+
+:::note
+No percent decoding or trimming of tabs, spaces or control characters is performed.
+:::
+)";
+    FunctionDocumentation::Syntax syntax_encode = "idnaEncode(s)";
+    FunctionDocumentation::Arguments arguments_encode = {
+        {"s", "Input string.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_encode = {"Returns an ASCII representation of the input string according to the IDNA mechanism of the input value.", {"String"}};
+    FunctionDocumentation::Examples examples_encode = {
+    {
+        "Usage example",
+        "SELECT idnaEncode('straße.münchen.de')",
+        R"(
+┌─idnaEncode('straße.münchen.de')─────┐
+│ xn--strae-oqa.xn--mnchen-3ya.de     │
+└─────────────────────────────────────┘
+        )"
+    }
+    };
+    FunctionDocumentation::IntroducedIn introduced_in = {24, 1};
+    FunctionDocumentation::Category category = FunctionDocumentation::Category::String;
+    FunctionDocumentation documentation_encode = {description_encode, syntax_encode, arguments_encode, returned_value_encode, examples_encode, introduced_in, category};
+
+    FunctionDocumentation::Description description_try_encode = R"(
+Returns the Unicode (UTF-8) representation (ToUnicode algorithm) of a domain name according to the [Internationalized Domain Names in Applications](https://en.wikipedia.org/wiki/Internationalized_domain_name#Internationalizing_Domain_Names_in_Applications) (IDNA) mechanism.
+In case of an error it returns an empty string instead of throwing an exception.
+)";
+    FunctionDocumentation::Syntax syntax_try_encode = "tryIdnaEncode(s)";
+    FunctionDocumentation::Arguments arguments_try_encode = {
+        {"s", "Input string.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_try_encode = {"Returns an ASCII representation of the input string according to the IDNA mechanism of the input value, or empty string if input is invalid.", {"String"}};
+    FunctionDocumentation::Examples examples_try_encode = {
+    {
+        "Usage example",
+        "SELECT tryIdnaEncode('straße.münchen.de')",
```

Comment:
I am pretty sure, I was the one who added the example in the first place :-)

Reply:
Ein hervorragendes Beispiel! :-)